specs/search/search.md

```diff
@@ -0,0 +1,128 @@
+
+### Prior Art/Examples
+
+- [Fluent UI - React](https://developer.microsoft.com/en-us/fluentui#/controls/web/searchbox)
+- [Material UI](https://material-ui.com/components/text-fields/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/input/)
+- [Ant Design](https://ant.design/components/input/)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/apps/design/controls/auto-suggest-box)
+
+---
+
+### API
+
+Extends FAST Element
+
+*Component Name*
+- `fast-search-box`
+
+*Attrs* - Adapted from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search)
+- `autofocus` - automatically focuses the control
+- `placeholder` - an exemplar value to display in the input field whenever it is empty
+- `label` - `aria-label` for the input when a <label> is not present
+- `required` - boolean value that sets the field as required
+- `value` - string value of the text field, can be an empty string
+- `disabled` - disables the control
+- `list` - the id of the `<datalist>` element that contains the optional pre-defined autocomplete options
+- `maxlength`	- the maximum number of characters the input should accept
+- `minlength` -	the minimum number of characters long the input can be and still be considered valid
+- `name` - the name of the control
+- `pattern` - a regular expression the input's contents must match in order to be valid
+- `size` - a number indicating how many characters wide the input field should be
+- `spellcheck` - controls whether or not to enable spell checking for the input field, or if the default spell checking configuration should be used
+.- `appearance` - enum
+  - outline (default)
+  - filled
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<label part="label"><slot></slot></label>
+<div part="root">
+  <slot name="before-content" part="before-content"></slot>
+  <input type="search" part="control" />
+  <slot name="clear-button" part="clear-button" aria-label="${x => x.label}"></slot>
+  <slot name="after-content" part="after-content"></slot>
+</div>
+<!-- end shadow root -->
+```
+
+
+*Screenshots below are of the basic appearance of the component and are not exhaustive.*
+
+| State | Image |
+| ----- | ----- |
+| default | ![](./images/search-box.png) |
+| search button | ![](./images/search-box-button.png)
+| focus | ![](./images/search-box-focus.png)
+| disabled | ![](./images/search-box-disabled.png)
+| custom icon | ![](./images/search-box-custom-icon.png)
+
+*Slot Names*
+- default - the label content
+- before-content - often times a glyph, icon, or button precedes input
+- after-content - often times a glyph, icon, or button follows the input
+- clear-content - often times a glyph, icon, or button follows that clears the input field
+
+*Host Classes*
+- disabled
```

Comment:
Can `disabled` and `required` just be attributes and not classes? It seems folks would still have styling hooks and we would avoid class-clobbering side-effects

Reply:
Right, thanks for pointing that out, This was modeled off of `TextField` Spec, I'll remove. Probably should also remove it there.

Reply:
yea that spec might be stale.